File: app-native_ios/app-native_ios/AuthManager.swift
//
//  AuthManager.swift
//  app-native_ios
//

import SwiftUI
import Combine

class AuthManager: ObservableObject {
    @Published var isLoggedIn = false
    @Published var username: String? = nil

    func login(username: String) {
        isLoggedIn = true
        self.username = username
    }

    func logout() {
        isLoggedIn = false
        username = nil
    }
}

File: app-native_android/app/src/main/java/com/incento/sdk_test_app/AuthManager.kt
package com.incento.sdk_test_app

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.ViewModel

class AuthManager : ViewModel() {
    var isLoggedIn by mutableStateOf(false)
        private set
    var username: String? by mutableStateOf(null)
        private set
    var selectedTab by mutableIntStateOf(0)
    var pendingWidgetOpen = false

    fun login(username: String) {
        isLoggedIn = true
        this.username = username.ifEmpty { "사용자" }
    }

    fun logout() {
        isLoggedIn = false
        username = null
    }
}
